mypy/ipc.py

```diff
@@ -0,0 +1,181 @@
+"""Cross platform abstractions for inter-process communication
+
+On Unix, this uses AF_UNIX sockets.
+On Windows, this uses NamedPipes.
+"""
+
+import contextlib
+import os
+import shutil
+import sys
+import tempfile
+
+from typing import Iterator, Optional, Callable
+
+MYPY = False
+if MYPY:
+    from typing import Type
+
+from types import TracebackType
+
+if sys.platform == 'win32':
+    # This may be private, but it is needed for IPC on Windows, and is basically stable
+    import _winapi
+    _IPCHandle = int
+else:
+    import socket
+    _IPCHandle = socket.socket
+
+
+class IPCException(Exception):
+    """Exception for IPC issues."""
+    pass
+
+
+class IPCBase:
+    """Base class for communication between the dmypy client and server.
+
+    This contains logic shared between the client and server, such as reading
+    and writing.
+    """
+
+    connection = None  # type: _IPCHandle
+
+    def __init__(self, name: str) -> None:
+        self.READ_SIZE = 100000
```

Comment:
Could you do a test where the errors are longer than the read size?

Reply:
Sure! I should probably add some integration tests for IPC anyway.

Reply:
I made a test that the IPC works as expected, including for when the message length is over the `READ_SIZE` (which is also over the maximum buffer size for named pipes).

Reply:
Actually, I seem to have an issue with unique connection names, so I will have to modify the test a bit.

Reply:
I actually just meant to go test it manually once, but this is even better!

Reply:
Well, I think it is valuable to have tests, and it appears they were worth it :)